### PR TITLE
chore(progressive-deployment): enable for elasticsearch-k8s-metrics-adapter

### PR DESCRIPTION
**Pull request by Z3r0Sum**
- Enable the helm chart bits for progressive deployments
  - Pass down deployment slice to QGs

### CODE REVIEW

File: helm/templates/_helpers.tpl [hunk 1]
```diff
@@ -27,3 +27,9 @@ Combined Pod labels
 {{- $new := merge .Values.podLabels $combinedLabels }}
 {{- toYaml $new }}
 {{- end -}}
+
+{{- define "engprod.deploymentslice" -}}
```
Comment (naemono) [inline helm/templates/_helpers.tpl:31]:
```suggestion
{{- define "elasticsearch-metrics-apiserver.deploymentslice" -}}
```
Making these definitions more consistent with all of the others.

File: helm/templates/deployment.yaml [hunk 2]
```diff
@@ -16,6 +16,9 @@ spec:
     metadata:
       labels:
         {{- include "elasticsearch-metrics-apiserver.combinedPodLabels" . | nindent 8 }}
+        {{- with (include "engprod.deploymentslice" . | trim) }}
```
Comment (naemono) [inline helm/templates/deployment.yaml:19]:
```suggestion
        {{- with (include "elasticsearch-metrics-apiserver.deploymentslice" . | trim) }}
```

**Review by naemono — APPROVED** [review]
Seems fine to me, but just wanted to make the helpers/definitions consistent.

File: helm/templates/_helpers.tpl (hunk 1 above)
Comment (Z3r0Sum) [reply helm/templates/_helpers.tpl:31]: @naemono This is really meant to be a generic `engprod` function going forward, so we can add other labels as necessary under our context.  Thoughts on leaving it named that way?

Comment (naemono) [reply helm/templates/_helpers.tpl:31]: > @naemono This is really meant to be a generic `engprod` function going forward, so we can add other labels as necessary under our context. Thoughts on leaving it named that way?

My thoughts for leaving it this way was that this is a public repo and it doesn't mean anything to anyone outside of elastic. but it's not a blocker to me.

Comment (Z3r0Sum) [reply helm/templates/_helpers.tpl:31]: That's fair. I can shift it.

**Pull request closed without merging**